src/temporal/t.rast.boxplot/t.rast.boxplot.py

```diff
@@ -1,631 +1,802 @@
+# %end
+
+# %option
+# % key: flier_marker
+# % type: string
+# % label: Flier marker
+# % description: Set flier marker (see https://matplotlib.org/stable/api/markers_api.html for options)
+# % required: no
+# % guisection: Boxplot format
+# %end
+
+# %option
+# % key: flier_size
+# % type: string
+# % label: Flier size
+# % description: Set the flier size
+# % required: no
+# % guisection: Boxplot format
+# %end
+
+# %option G_OPT_C
+# % key: flier_color
+# % label: Flier color
+# % description: Set the flier color
+# % required: no
+# % guisection: Boxplot format
+# %end
+
+
+import atexit
+import sys
+import uuid
+from subprocess import PIPE
+
+from dateutil import parser
+
+import matplotlib.dates as mdates
+
+import grass.script as gs
+from grass.pygrass.modules import Module
+
+
+clean_maps = []
+
+
+def create_unique_name(name):
+    """Generate a temporary name which contains prefix
+    Store the name in the global list.
+
+    :param str name: prefix to be used for unique string
+
+    :return str: Unique string with user defined prefix
+    """
+    unique_string = f"{name}{uuid.uuid4().hex}"
+    return unique_string
+
+
+def create_temporary_name(prefix):
+    """Create temporary file name and add this to clean_maps
+
+    :param str name: prefix to be used for unique string
+
+    :return str: Unique string with user defined prefix
+    """
+    tmpf = create_unique_name(prefix)
+    clean_maps.append(tmpf)
+    return tmpf
+
+
+def cleanup():
+    """Remove temporary maps specified in the global list"""
+    maps = reversed(clean_maps)
+    mapset = gs.gisenv()["MAPSET"]
+    for map_name in maps:
+        for element in ("raster", "vector"):
+            found = gs.find_file(
+                name=map_name,
+                element=element,
+                mapset=mapset,
+            )
+            if found["file"]:
+                Module(
+                    "g.remove",
+                    flags="f",
+                    type=element,
+                    name=map_name,
+                    quiet=True,
+                )
+
+
+def strip_mapset(name, join_char="@"):
+    """Strip Mapset name and '@' from map name
+    >>> strip_mapset('elevation@PERMANENT')
+    elevation
+
+    :param str name: map name
+    :param str join_char: Character separating map and mapset name
+
+    :return str: mapname without the mapset name
+    """
+    return name.split(join_char)[0] if join_char in name else name
+
+
+def get_range(option_whisker_range):
+    """Get and apply range for whisker (multiplication factor)
+
+    :param str option_whisker_range: the addon parameter 'range'
+
+    :return float: number representing the whisker range
+    """
+    if option_whisker_range:
+        whisker_range = float(option_whisker_range)
+    else:
+        whisker_range = 1.5
+    if whisker_range <= 0:
+        gs.fatal("The range value need to be larger than 0")
+    return whisker_range
+
+
+def get_rast_name_dates(option_input):
```

Comment:
Simpliefied version with use this func as `parse` param argument in `gs.paser_command()` fuc, comment bellow, check the line 698.

```python
def get_rast_name_dates(rasters, col_sep):
    """Create list of names, tic positions, dates and temporal type
    of raster layers in input strds

    :param str rasters: rasters (output of the t.rast.list module)
    :param str col_sep: column separator e.g. "2000_01_tempmean|
                                               2000-01-01 00:00:00|absolute"

    :return tuple: tuple with list of dates, list of raster names,
                         list of tic positions and temporal type of
                         raster layers
    """
    raster_names = []
    raster_dates = []
    temp_types = set()
    tic_positions = []
    for raster in rasters.splitlines():
        raster_name, raster_date, temp_type = raster.split(sep)
        raster_names.append(raster_name)
        raster_dates.append(raster_date)
        temp_types.add(temp_type)
        if temp_type == "relative":
            tic_positions.append(int(raster_date))
        else:
            tic_positions.append(parser.parse(timestr=raster_date))
    return (raster_names, raster_dates, tic_positions, list(temp_types)[0])
```